DesignDoc.md

```diff
@@ -0,0 +1,214 @@
+
+2. As a user, I want to be able to search for movies by title, director, or actor so that I can easily find the films I'm looking for.
+
+3. As a user, I want to be able to rate and review movies on Reel Deal so that I can share my thoughts with others.
+
+4. As a user, I want to be able to follow other users on Reel Deal so that I can see their reviews and ratings, and discover new movies.
+
+5. As a user, I want to be able to save movies to my "watchlist" on Reel Deal so that I can keep track of the films I want to watch later.
+
+6. As a user, I want to be able to create custom lists on Reel Deal so that I can organize my favorite films by genre, director, or any other criteria.
+
+7. As a user , I want to be able to view statistics and data on movies on Reel Deal so that I can learn more about the films I watch.
+
+8. As a user, I want to be able to view trailers and posters for movies on Reel Deal so that I can get a better idea of what a film is about before watching it.
+
+9. As a user, I want to be able to connect my ReelDeal account to my social media accounts so that I can share my reviews and ratings with my friends.
+
+### 4.1. In Scope
+
+* looking at reviews from others people
+* search movies by title director and actor
+* show popular movies to leave reviews on
+* Adding, updating, deleting, and viewing reviews
+
+
+### 4.2. Out of Scope
+
+* following other users 
+* login with socials
+
+# 5. Proposed Architecture Overview
+
+
+We will use API Gateway and Lambda to create endpoints that will handle the creation, updating, and retrieval of movies and reviews. 
+
+We will store movie review information in a table in DynamoDB.
+
+ReelDeal will also provide a web interface for users. A main page providing a variety of options to add, update and reminisce of all the movies you watched and maybe some you warn others not to watch.
+
+
+# 6. API
+
+## 6.1. Public Models
+
+```
+// Movie 
+    private int id;
+    private String title;
+    private String description;
+    private String releaseDate;
+    private String posterUrl;
+    private List<String> genres;
+    private List<String> cast;
+    private String director;
+
+
+// Review 
+    private int id;
+    private int movieId;
+    private String username;
+    private String text;
+    private int rating;
+    private Date date;
+
+
+// User 
+    private int id;
+    private String username;
+    private String password;
+    private String email;
+    private List<Review> reviews;
+
+
+```
+Get All Movies Endpoint
+GET /movies - Retrieve a list of all movies
+if there is no data, will throw NoDataFoundException
+
+Get Movie Endpoint
+GET /movies/{id} - Retrieve a specific movie by its ID
+if movie id is not found throw 'MovieNotFoundException'
+
+Create Movie Endpoint
+POST /movies - Accepts post to create a new movie includes title and short description for non null fields 
+if movie contains invalid characters will throw 'InvalidAttributeValueException'
+{ 
```

Comment:
These look good. You should wrap them all in triple backticks so that they render correctly though - as-is this is pretty hard to read when viewing the file (i.e. not in the PR).